### PR TITLE
Add demo jupyter notebooks and update test workflow

### PR DESCRIPTION
**Pull request by sfc-gh-lmukhopadhyay**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1373899

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Updates notebook testing workflow with increased cell timeout,  and adds SnowparkPandasAPIDemo.ipynb notebook from customer demo and SnowflakeChainTesting.ipynb which was previously blocked.

### CODE REVIEW

**Review by sfc-gh-mvashishtha — APPROVED** [review]
LGTM pending reverting the YML to make the tests run daily instead of on `pull_request`. Would you please ping me once you have all approvals and that's done?

**Review by sfc-gh-mvashishtha — APPROVED** [review]
LGTM pending reverting the YML to make the tests run daily instead of on `pull_request`. Would you please ping me once you have all approvals and that's done?

**Review by sfc-gh-nkrishna — APPROVED** [review]

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/daily_jupyter_nb_test.yml
Comment (sfc-gh-mvashishtha) [inline .github/workflows/daily_jupyter_nb_test.yml:68]: Do we want to limit the test to just AWS?

**Review by sfc-gh-mvashishtha — APPROVED** [review]
one optional question. otherwise LGTM

File: .github/workflows/daily_jupyter_nb_test.yml
Comment (sfc-gh-lmukhopadhyay) [reply .github/workflows/daily_jupyter_nb_test.yml:68]: The FINANCIAL__ECONOMIC_ESSENTIALS.CYBERSYN database used for the Snowpark Pandas demo wasn't available in azure or gcp so I had to limit the test to aws for now

Comment (sfc-gh-mvashishtha) [reply .github/workflows/daily_jupyter_nb_test.yml:68]: could you add a comment about that?

i doubt we need to test on the other cloud providers anyway, but it's good for future reference to have that information here.

Comment (sfc-gh-lmukhopadhyay) [reply .github/workflows/daily_jupyter_nb_test.yml:68]: add a comment in the yaml file? Sure

**Review by sfc-gh-mvashishtha — APPROVED** [review]

**Pull request merged**
